### PR TITLE
[CFP-137] Signpost unused material fees on claim summary

### PR DESCRIPTION
**Pull request by jrmhaig**
#### What

Add a notice on the claim summary for qualifying claims to indicate that an unused material fee may be claimed.

#### Ticket

[Unused material fee flag on summary page](https://dsdmoj.atlassian.net/browse/CFP-137)

#### Why

Providers are being underpaid for work they are entitled to claim for.

#### How

Following on from #4011, add the notices to the summary page based on the same rules.

### CODE REVIEW

File: app/views/external_users/claims/summary.html.haml [hunk 1]
```diff
@@ -6,6 +6,15 @@
 
 - present(@claim) do |claim|
 
+  - if display_unused_materials_notice_for(claim)
+    = govuk_warning_text(t('external_users.claims.misc_fees.unused_materials.heading')) do
+      = t('external_users.claims.misc_fees.unused_materials.notice')
+      %br
+      = t('external_users.claims.misc_fees.unused_materials.hint_start')
+      = govuk_link_to t('external_users.claims.misc_fees.unused_materials.hint_link'), edit_polymorphic_path(claim, step: :miscellaneous_fees, referrer: :summary)
+      = t('external_users.claims.misc_fees.unused_materials.hint_end')
```
Comment (jsugarman) [inline app/views/external_users/claims/summary.html.haml]: is it possible to combine the hint parts into one html translation

```
misc_fee_link = govuk_link_to t('external_users.claims.misc_fees.unused_materials.hint_link'), edit_polymorphic_path(claim, step: :miscellaneous_fees, referrer: :summary)
t('external_users.claims.misc_fees.unused_materials.hint_html', link: misc_fee_link))
```

and translation:

```
hint_html: "Goto %{link} to add this to your claim"
```

File: spec/helpers/claims_helper_spec.rb [hunk 2]
```diff
@@ -202,4 +202,28 @@ def current_user
...
+
+  describe '#misc_fees_locals' do
+    subject(:locals) { misc_fees_locals(claim) }
+
+    let(:claim) { build :claim }
+
+    context 'with a claim eligible for unused materials fees' do
+      before { allow(claim).to receive(:unused_materials_applicable?).and_return true }
+
+      it { expect(locals[:unclaimed_fees_notice]).not_to be_nil }
```
Comment (jsugarman) [inline spec/helpers/claims_helper_spec.rb]: prefer positive tests if possible plus you could test the value which would have the benefit of testing the translation exists and has the expected content.
```
it { expect(locals.keys).to include(:unclaimed_fees_notice) }
it { expect(locals[:unclaimed_fees_notice]).to eql('blah'))
```

File: lib/govuk_component/warning_text_helpers.rb [hunk 3]
```diff
@@ -2,15 +2,25 @@
...
       text__assistive = tag.span(assistive_text, class: 'govuk-warning-text__assistive')
       content = sanitize(body)
       tag.div(**tag_options) do
         concat tag.span('!', class: 'govuk-warning-text__icon', 'aria-hidden': true)
         concat tag.strong(text__assistive + content, class: 'govuk-warning-text__text')
+        govuk_warning_text_description(&block)
       end
     end
+
+    def govuk_warning_text_description(&block)
```
Comment (jsugarman) [inline lib/govuk_component/warning_text_helpers.rb:17]: needs tests

File: lib/govuk_component/warning_text_helpers.rb [hunk 4]
```diff
@@ -2,15 +2,25 @@
 
 module GovukComponent
   module WarningTextHelpers
-    def govuk_warning_text(body = nil, assistive_text = t('common.warning'), **tag_options)
+    def govuk_warning_text(body = nil, assistive_text = t('common.warning'), **tag_options, &block)
```
Comment (jsugarman) [inline lib/govuk_component/warning_text_helpers.rb:5]: needs tests

File: config/locales/en.yml [hunk 5]
```diff
@@ -1374,7 +1374,13 @@ en:
           <a href="https://www.gov.uk/government/publications/af1-claim-for-advocate-graduated-fees" class="govuk-link" rel="noreferrer noopener" target="_blank">appropriate special prep form <span class="govuk-visually-hidden">(opens in new tab)</span></a>
           or <a href="https://www.gov.uk/government/publications/af1-claim-for-advocate-graduated-fees" class="govuk-link" rel="noreferrer noopener" target="_blank">unused materials (over 3 hours) form <span class="govuk-visually-hidden">(opens in new tab)</span></a>
           and attach to the claim.
-        unused_materials_notice: This claim should be eligible for unused materials fees (up to 3 hours)
+        unused_materials:
+          heading: Unclaimed fees
+          notice: This claim should be eligible for unused materials fees (up to 3 hours)
+          notice_long: This claim should be eligible for unused materials fees (up to 3 hours) but they haven't been claimed
```
Comment (jsugarman) [inline config/locales/en.yml]: will require tweaking previous work but you could 
```
notice:
  short: This claim should be eligible for unused materials fees (up to 3 hours)
  long: This claim should be eligible for unused materials fees (up to 3 hours) but they haven't been claimed
```

File: app/helpers/claims_helper.rb [hunk 6]
```diff
@@ -46,7 +46,19 @@ def fee_shared_headings(claim, scope)
       page_header: t('page_header', scope: scope),
       page_hint: t('page_hint', scope: scope)
     }.tap do |headings|
-      headings[:page_notice] = t('unused_materials_notice', scope: scope) if display_unused_materials_notice_for(claim)
+      headings[:page_notice] = t('unused_materials.notice', scope: scope) if display_unused_materials_notice_for(claim)
+    end
+  end
+
+  def misc_fees_locals(claim, args = {})
```
Comment (jsugarman) [inline app/helpers/claims_helper.rb]: 🤔 not too sure about this. It extracts logic from the view which is nice but: 
1. this partial is small and specific anyway and this add adds a layer of indirection
2. this helper is specific to misc fee **summary** but it exists in claims helper so could be confused with generic misc fees.

Comment (jsugarman) [reply app/helpers/claims_helper.rb]: options would seem to be 
1. put conditions in partial
2. name change to indicate summary
3. perhaps we need a new claim_summary_helper generally

**Review by jsugarman — CHANGES_REQUESTED** [review]

File: lib/govuk_component/warning_text_helpers.rb (hunk 4 above)
Comment (jrmhaig) [reply lib/govuk_component/warning_text_helpers.rb:5]: There is a test for this but I may not have checked it in when you reviewed this.

File: app/helpers/claims_helper.rb (hunk 6 above)
Comment (jrmhaig) [reply app/helpers/claims_helper.rb]: I am renaming it to `misc_fees_summary_locals`

**Comment by jrmhaig** [comment]
Replaced with #4042

**Pull request closed without merging**